magento/ece-tools
# MCLOUD-5665: Improve git installation add support for packages with different structure

### Description
https://jira.corp.magento.com/browse/MCLOUD-5665

### Manual testing scenarios

1. Deploy Magento from git to cloud environment
https://wiki.corp.magento.com/pages/viewpage.action?spaceKey=MCP&title=ECE+Tools+-+Simple+Flow+to+Deploy+Magento+to+ECE+from+GitHub+Branch
In step 4 add two additional packages
```
stage:
  global:
    DEPLOYED_MAGENTO_VERSION_FROM_GIT: '2.3.3'
    DEPLOY_FROM_GIT_OPTIONS:
      clear_magento_module_requirements: true
      repositories:
        ....
        inventory:
           branch: <branch>
           repo: https://<token>@github.com/magento/inventory.git
           type: flat-structure
        security-package:
           branch:  <branch>
           repo: https://<token>@github.com/magento/security-package.git
           type: flat-structure
```
2. Check that modules from `inventory` and `security-package` were added to generated composer.json and are installed on cloud.

### Contribution checklist
